src/senaite/core/content/sampletype.py

```diff
@@ -0,0 +1,470 @@
+            default=u"Default Container Type"
+        ),
+        description=_(
+            u"description_sampletype_containertype",
+            default=u"The default container type. New sample partitions "
+                    u"are automatically assigned a container of this "
+                    u"type, unless it has been specified in more details "
+                    u"per analysis service"
+        ),
+        allowed_types=("ContainerType", ),
+        multi_valued=False,
+        required=False,
+    )
+
+    # Stickers (Data Grid)
+    directives.widget(
+        "admitted_sticker_templates",
+        DataGridWidgetFactory,
+        allow_insert=False,
+        allow_delete=False,
+        allow_reorder=False,
+        auto_append=False)
+    admitted_sticker_templates = DataGridField(
+        title=_(u"label_sampletype_admitted_stickers_templates",
+                default=u"Admitted sticker templates"),
+        description=_(u"description_sampletype_admitted_stickers_templates",
+                      default=u"Defines the stickers to use for "
+                              u"this sample type."),
+        value_type=DataGridRow(schema=IStickersRecordSchema),
+        required=True,
+        default=[{
+            'admitted': set(),
+            'small_default': None,
+            'large_default': None
+        }]
+    )
+
+
+class StickersFieldValidator(validator.SimpleFieldValidator):
+    """Custom validator for DGF
+    """
+
+    def validate(self, value):
+        if value is None:
+            value = []
+
+        _valid = False
+        _msg = _(u"Invalid sticker object format")
+
+        if len(value) == 1:
+            _valid = True
+            _errors = []
+            if not len(value[0]['admitted']):
+                _valid = False
+                _errors.append(
+                    _(u'at least one admitted sticker must be chosen'))
+            if not value[0]['small_default']:
+                _valid = False
+                _errors.append(_(u'select small default sticker'))
+            if not value[0]['large_default']:
+                _valid = False
+                _errors.append(_(u'select large default sticker'))
+            _msg = _(u"ERRORS: ") + ", ".join(_errors)
+
+        if not _valid:
+            raise Invalid(_msg)
+
+
+validator.WidgetValidatorDiscriminators(
+    StickersFieldValidator,
+    field=ISampleTypeSchema["admitted_sticker_templates"],
+)
+
+
+@implementer(ISampleType, ISampleTypeSchema, IDeactivable)
+class SampleType(Container, SampleTypeAwareMixin):
+    """SampleType
+    """
+    # Catalogs where this type will be catalogued
+    _catalogs = [SETUP_CATALOG]
+
+    security = ClassSecurityInfo()
+
+    @security.protected(permissions.View)
+    def getRawRetentionPeriod(self):
+        accessor = self.accessor("retention_period")
+        return accessor(self)
+
+    @security.protected(permissions.View)
+    def getRetentionPeriod(self, raw=False):
```

Comment:
As far as I am concerned the parameter `raw` is not used anywhere. And since we already have the function `getRawRetentionPeriod`, we can simply remove this parameter and let `getRetentionPeriod` function to directly return the `dict` representation of the `Timedelta`.

On another note (future, not necessary for this PR), we might consider this function to directly return the value (a `timedelta`) and do the conversion to the `dict` representation elsewhere, by means of a new function in `senaite.core.api.dtime`, e.g.:

```python
def to_delta_dict(value, default=_marker):
    if not isinstance(value, Timedelta):
        raise TypeError("%r is not supported" % value)

    # Note timedelta keeps days and seconds a part!
    return {
        "days": value.days,
        "hours": (value.seconds / 3600) % 24, # hours within a day
        "minutes": (value.seconds / 60) % 60, # minutes within an hour
        "seconds": value.seconds % 60, # seconds within a minute
    }
```

We could then rely on this function for the conversion [in the sample types view](https://github.com/senaite/senaite.core/pull/2584/files#diff-7426b56b15aa1f4b65ae9229a7c881d59a72510b0a7ba898f12c056d2cfc623cR158) in the [sample add form](https://github.com/senaite/senaite.core/blob/5cd7ad334aa4cb41153ef1e98a82d9b7b005b181/src/bika/lims/browser/analysisrequest/add2.py#L989), and [in the widget as well](https://github.com/toropok/senaite.core/blob/dx-sampletype/src/senaite/core/z3cform/widgets/duration/widget.py#L41).